### ISSUE TITLE
Drag’n’Drop multiple blocks in a column breaks the editor

### ISSUE THREAD

**Issue by MatzeKitt** [WordPress/gutenberg#34631]
### Description

If you drag multiple blocks inside a column block, it results in a broken editor with the following error message:

```
pp@http://localhost/wp-clean/wp-includes/js/dist/block-editor.min.js?ver=390a9f83b606a55e8b523c4d16bdb08c:12:393567
@http://localhost/wp-clean/wp-includes/js/dist/block-editor.min.js?ver=390a9f83b606a55e8b523c4d16bdb08c:12:394549
Bh@http://localhost/wp-clean/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:126:457
Dj@http://localhost/wp-clean/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:162:478
unstable_runWithPriority@http://localhost/wp-clean/wp-includes/js/dist/vendor/react.min.js?ver=16.13.1:25:261
Te@http://localhost/wp-clean/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:146:81
Te@[native code]
@http://localhost/wp-clean/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:61:69
unstable_runWithPriority@http://localhost/wp-clean/wp-includes/js/dist/vendor/react.min.js?ver=16.13.1:25:261
Pg@http://localhost/wp-clean/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:61:16
ha@http://localhost/wp-clean/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:60:453
Ja@http://localhost/wp-clean/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:224:402
ch@http://localhost/wp-clean/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:93:81
ch@[native code]
e@http://localhost/wp-clean/wp-includes/js/dist/data.min.js?ver=7bfa3a673439e85e459d5ec92d30fdda:2:20892
t@http://localhost/wp-clean/wp-includes/js/dist/data.min.js?ver=7bfa3a673439e85e459d5ec92d30fdda:2:20959
@http://localhost/wp-clean/wp-includes/js/dist/data.min.js?ver=7bfa3a673439e85e459d5ec92d30fdda:2:15083
b@http://localhost/wp-clean/wp-includes/js/dist/data.min.js?ver=7bfa3a673439e85e459d5ec92d30fdda:2:8589
@http://localhost/wp-clean/wp-includes/js/dist/redux-routine.min.js?ver=7f20dbcdb5d5645c977063e356eb95cf:2:1354
some@[native code]
r@http://localhost/wp-clean/wp-includes/js/dist/redux-routine.min.js?ver=7f20dbcdb5d5645c977063e356eb95cf:2:9237
@http://localhost/wp-clean/wp-includes/js/dist/redux-routine.min.js?ver=7f20dbcdb5d5645c977063e356eb95cf:2:9187
@http://localhost/wp-clean/wp-includes/js/dist/redux-routine.min.js?ver=7f20dbcdb5d5645c977063e356eb95cf:2:1321
some@[native code]
r@http://localhost/wp-clean/wp-includes/js/dist/redux-routine.min.js?ver=7f20dbcdb5d5645c977063e356eb95cf:2:9237
@http://localhost/wp-clean/wp-includes/js/dist/redux-routine.min.js?ver=7f20dbcdb5d5645c977063e356eb95cf:2:9187
@http://localhost/wp-clean/wp-includes/js/dist/redux-routine.min.js?ver=7f20dbcdb5d5645c977063e356eb95cf:2:1321
some@[native code]
r@http://localhost/wp-clean/wp-includes/js/dist/redux-routine.min.js?ver=7f20dbcdb5d5645c977063e356eb95cf:2:9237
@http://localhost/wp-clean/wp-includes/js/dist/redux-routine.min.js?ver=7f20dbcdb5d5645c977063e356eb95cf:2:9187
c@http://localhost/wp-clean/wp-includes/js/dist/redux-routine.min.js?ver=7f20dbcdb5d5645c977063e356eb95cf:2:9288
t@http://localhost/wp-clean/wp-includes/js/dist/redux-routine.min.js?ver=7f20dbcdb5d5645c977063e356eb95cf:2:9544
Promise@[native code]
@http://localhost/wp-clean/wp-includes/js/dist/redux-routine.min.js?ver=7f20dbcdb5d5645c977063e356eb95cf:2:1416
@http://localhost/wp-clean/wp-includes/js/dist/data.min.js?ver=7bfa3a673439e85e459d5ec92d30fdda:2:13326
@http://localhost/wp-clean/wp-includes/js/dist/block-editor.min.js?ver=390a9f83b606a55e8b523c4d16bdb08c:12:239116
@http://localhost/wp-clean/wp-includes/js/dist/block-editor.min.js?ver=390a9f83b606a55e8b523c4d16bdb08c:12:239914
h@http://localhost/wp-clean/wp-includes/js/dist/compose.min.js?ver=6284a4846a73f9c3e3ff2d7bd0b47be7:2:13597
```

This symptom is similar to #34409.

### Step-by-step reproduction instructions

1. Create this structure:

```html
<!-- wp:heading -->
<h2>Überschrift</h2>
<!-- /wp:heading -->

<!-- wp:paragraph -->
<p>Weit hinten, hinter den Wortbergen, fern der Länder Vokalien und Konsonantien leben die Blindtexte. Abgeschieden wohnen sie in Buchstabhausen an der Küste des Semantik, eines großen Sprachozeans. Ein kleines Bächlein namens Duden fließt durch ihren Ort und versorgt sie mit den nötigen Regelialien. Es ist ein paradiesmatisches Land, in dem einem gebratene Satzteile in den Mund fliegen. Nicht einmal von der allmächtigen Interpunktion werden die Blindtexte beherrscht – ein geradezu unorthographisches</p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p>Weit hinten, hinter den Wortbergen, fern der Länder Vokalien und Konsonantien leben die Blindtexte. Abgeschieden wohnen sie in Buchstabhausen an der Küste des Semantik, eines großen Sprachozeans. Ein kleines Bächlein namens Duden fließt durch ihren Ort und versorgt sie mit den nötigen Regelialien. Es ist ein paradiesmatisches Land, in dem einem gebratene Satzteile in den Mund fliegen. Nicht einmal von der allmächtigen Interpunktion werden die Blindtexte beherrscht – ein geradezu unorthographisches</p>
<!-- /wp:paragraph -->

<!-- wp:columns -->
<div class="wp-block-columns"><!-- wp:column -->
<div class="wp-block-column"></div>
<!-- /wp:column -->

<!-- wp:column -->
<div class="wp-block-column"></div>
<!-- /wp:column --></div>
<!-- /wp:columns -->
````

2. Select the two paragraph blocks and drag’n’drop them into the first column of the columns block.

### Screenshots, screen recording, code snippet

https://user-images.githubusercontent.com/1018979/132449887-f68d228f-3df0-4d78-8632-48cf02a6e753.mp4

### Environment info

– WordPress 5.8
– No Gutenberg plugin installed
– Safari 14/Chrome 92
– macOS 11.5

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by Mamaduka** [comment]
Hi, @MatzeKitt

I can reproduce the issue in WP 5.8, but I'm no longer able to reproduce it with Gutenberg 11.4.

Original error message:

```
Uncaught TypeError: Cannot read properties of null (reading 'firstChild') at getDeepestNode
```

The error was coming from the `useMultiSelection` hook. We recently updated this hook to fix the different errors, and it looks like we also fixed this issue. See #33896.

I would appreciate your confirmation that the issue no longer exists in the latest stable Gutenberg plugin.

Thanks

**Comment by MatzeKitt** [comment]
Hi @Mamaduka 

Unfortunately, I can also reproduce it with Gutenberg 11.4.1, which I installed just now.

The error message:

```
Ps@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/block-editor/index.min.js?ver=5811b9f99a23806bdb23e7b21b14d794:11:17688
@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/block-editor/index.min.js?ver=5811b9f99a23806bdb23e7b21b14d794:11:18576
@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/compose/index.min.js?ver=0f1f9e9ebb624d4dfb57670a95cb1653:1:22834
A@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/compose/index.min.js?ver=0f1f9e9ebb624d4dfb57670a95cb1653:1:24940
@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/compose/index.min.js?ver=0f1f9e9ebb624d4dfb57670a95cb1653:1:25205
forEach@[native code]
@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/compose/index.min.js?ver=0f1f9e9ebb624d4dfb57670a95cb1653:1:25155
commitHookEffectListMount@http://localhost/wp-clean/wp-content/plugins/gutenberg/vendor/react-dom.860550b4.js:20603:34
commitLifeCycles@http://localhost/wp-clean/wp-content/plugins/gutenberg/vendor/react-dom.860550b4.js:20664:38
commitLayoutEffects@http://localhost/wp-clean/wp-content/plugins/gutenberg/vendor/react-dom.860550b4.js:23456:25
callCallback@http://localhost/wp-clean/wp-content/plugins/gutenberg/vendor/react-dom.860550b4.js:3942:21
dispatchEvent@[native code]
invokeGuardedCallbackDev@http://localhost/wp-clean/wp-content/plugins/gutenberg/vendor/react-dom.860550b4.js:3991:31
invokeGuardedCallback@http://localhost/wp-clean/wp-content/plugins/gutenberg/vendor/react-dom.860550b4.js:4053:38
commitRootImpl@http://localhost/wp-clean/wp-content/plugins/gutenberg/vendor/react-dom.860550b4.js:23181:32
commitRootImpl@[native code]
unstable_runWithPriority@http://localhost/wp-clean/wp-content/plugins/gutenberg/vendor/react.c58d0bd5.js:2942:26
commitRoot@http://localhost/wp-clean/wp-content/plugins/gutenberg/vendor/react-dom.860550b4.js:23020:22
performSyncWorkOnRoot@http://localhost/wp-clean/wp-content/plugins/gutenberg/vendor/react-dom.860550b4.js:22359:15
performSyncWorkOnRoot@[native code]
@http://localhost/wp-clean/wp-content/plugins/gutenberg/vendor/react-dom.860550b4.js:11357:36
unstable_runWithPriority@http://localhost/wp-clean/wp-content/plugins/gutenberg/vendor/react.c58d0bd5.js:2942:26
flushSyncCallbackQueueImpl@http://localhost/wp-clean/wp-content/plugins/gutenberg/vendor/react-dom.860550b4.js:11352:28
flushSyncCallbackQueue@http://localhost/wp-clean/wp-content/plugins/gutenberg/vendor/react-dom.860550b4.js:11339:31
scheduleUpdateOnFiber@http://localhost/wp-clean/wp-content/plugins/gutenberg/vendor/react-dom.860550b4.js:21923:33
dispatchAction@http://localhost/wp-clean/wp-content/plugins/gutenberg/vendor/react-dom.860550b4.js:16169:28
dispatchAction@[native code]
e@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/data/index.min.js?ver=da4272fbc833ab25871c55f89400d555:1:19842
t@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/data/index.min.js?ver=da4272fbc833ab25871c55f89400d555:1:19909
@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/data/index.min.js?ver=da4272fbc833ab25871c55f89400d555:1:14313
@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/data/index.min.js?ver=da4272fbc833ab25871c55f89400d555:1:13435
y@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/data/index.min.js?ver=da4272fbc833ab25871c55f89400d555:1:6847
@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/redux-routine/index.min.js?ver=7a21a7d0e50a3686a159176880d88a37:1:8354
some@[native code]
n@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/redux-routine/index.min.js?ver=7a21a7d0e50a3686a159176880d88a37:1:3760
@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/redux-routine/index.min.js?ver=7a21a7d0e50a3686a159176880d88a37:1:3710
@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/redux-routine/index.min.js?ver=7a21a7d0e50a3686a159176880d88a37:1:8307
some@[native code]
n@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/redux-routine/index.min.js?ver=7a21a7d0e50a3686a159176880d88a37:1:3760
@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/redux-routine/index.min.js?ver=7a21a7d0e50a3686a159176880d88a37:1:3710
@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/redux-routine/index.min.js?ver=7a21a7d0e50a3686a159176880d88a37:1:8307
some@[native code]
n@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/redux-routine/index.min.js?ver=7a21a7d0e50a3686a159176880d88a37:1:3760
@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/redux-routine/index.min.js?ver=7a21a7d0e50a3686a159176880d88a37:1:3710
c@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/redux-routine/index.min.js?ver=7a21a7d0e50a3686a159176880d88a37:1:3811
r@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/redux-routine/index.min.js?ver=7a21a7d0e50a3686a159176880d88a37:1:4067
Promise@[native code]
@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/redux-routine/index.min.js?ver=7a21a7d0e50a3686a159176880d88a37:1:8413
@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/data/index.min.js?ver=da4272fbc833ab25871c55f89400d555:1:11692
@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/block-editor/index.min.js?ver=5811b9f99a23806bdb23e7b21b14d794:53:5271
@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/block-editor/index.min.js?ver=5811b9f99a23806bdb23e7b21b14d794:53:5765
h@http://localhost/wp-clean/wp-content/plugins/gutenberg/build/compose/index.min.js?ver=0f1f9e9ebb624d4dfb57670a95cb1653:1:30531
```

The original error in the console:

```
[Error] TypeError: null is not an object (evaluating 'e[n]')
```

**Comment by Mamaduka** [comment]
Thanks for the follow-up, @MatzeKitt.

I will add the "Needs Testing" label back and will tests during the Triage next week with other contributors.

**Comment by camilopbam** [comment]
I'm getting this very same issue in WP 5.8.1 with both the columns block and other custom made layout blocks.

**Comment by Mamaduka** [comment]
I just merged a fix for this issue, and it should be available in Gutenberg 11.7.
